compiler/codegen/cg-expr.cpp

```diff
@@ -2188,7 +2188,9 @@ static GenRet codegenSqrt(GenRet a) {
   GenRet ret;
   if (a.chplType && a.chplType->symbol->isRefOrWideRef()) a = codegenDeref(a);
   GenRet av = codegenValue(a);
-  if (info->cfile) {
+  if (info->cfile ||
+      (usingGpuLocaleModel() &&
+       getGpuCodegenType() == GpuCodegenType::GPU_CG_NVIDIA_CUDA)) {
```

Comment:
Expanding this to AMD does hurt performance even on newer GPUs. Strange, but we should keep this only for NVIDIA.

My remaining question is whether we should expand it to abs, too. I think we should just to be on the safe side, but I don't have a benchmark for abs performance. If you end up doing that, it'd be great if you can put this bool logic into a small helper.

`preferCMathOverLLVMIntr` or something could be the name of the helper.

Reply:
I agree, in the interest of doing no harm prior to the release I will add `preferCMathOverLLVMIntr` for abs as well. But this is definitely something we should look into post-release